### PR TITLE
Adds in-app Survey loading view

### LINKED ISSUE #2571: Consider adding a loading indicator while the survey loads
@Jaclyn mentioned that the survey takes around 1-2s to load.

We should consider adding a loading indicator if that improves the user experience

### PR DESCRIPTION
**Pull request by Ecarrion**
fixes #2571 

# Why

Prior to this PR, the user was left staring at an empty screen while the in-app survey loads. Since the survey can take a couple of seconds to load, we have decided to add a loading indicator view.

# How

- Creates a new `SuveryLoadingView` to hold the loading indicator
- Present the loading view when the request to load the survey is fired
- Hide the loading view when the survey has finished loading

PS: I've also updated some tests to use the new method name case.

# GIF

Survey-loading-view

# Testing Steps

- Add the following code in `InAppFeedbackCardViewController`, `configureDidNotLikeButton()` method
```swift
didNotLikeButton.on(.touchUpInside) { [weak self] _  in
    let vc = SurveyViewController(survey: .inAppFeedback, onCompletion: { })
    let nav = WooNavigationController(rootViewController: vc)
    self?.present(nav, animated: true)
}
```
- Tap on the "Could be better" button in the feedback card on the dashboard
- See that the loading spinner is present while the survey loads.

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Survey/SurveyLoadingView.swift [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+private extension SurveyLoadingView {
+    enum Localization {
+        static let wait = NSLocalizedString("Please wait", comment: "Text on the loading view of the survey screen indicating the user to wait")
+    }
+
+    enum Layout {
+        static let stackViewEdges = UIEdgeInsets(top: 38, left: 54, bottom: 48, right: 54)
+        static let stackViewSpacing = CGFloat(40)
+        static let indicatorScaleFactor = CGFloat(2.5)
+        static let corderRadius = CGFloat(14.5)
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Survey/SurveyLoadingView.swift:74]: nit: corderRadius -> cornerRadius :)

File: WooCommerce/Classes/ViewRelated/Survey/SurveyViewController.swift [hunk 2]
```diff
@@ -41,6 +46,20 @@ final class SurveyViewController: UIViewController {
...
+    private func addLoadingView() {
+        loadingView.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(loadingView)
+        view.pinSubviewAtCenter(loadingView)
+    }
+
+    /// Removes the loading view from the screen with a fade out animaition
+    ///
+    private func removeLoadingView() {
+        loadingView.fadeOut()
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Survey/SurveyViewController.swift:61]: nit: not sure if we might want to remove `loadingView` from its superview on fade out completion?

File: WooCommerce/WooCommerceTests/ViewRelated/Survey/SurveyViewControllerTests.swift [hunk 3]
```diff
@@ -37,13 +35,14 @@ final class SurveyViewControllerTests: XCTestCase {
...
 
         // Then
-        XCTAssertTrue(surveyCompleted)
+        waitUntil(timeout: 1) {
+            surveyCompleted
+        }
     }
 
-    func testItDoesNotCompletesAfterReceivingAFormSubmittedGETCallbackRequest() throws {
+    func test_it_does_not_completes_after_receiving_a_form_submitted_GET_callback_request() throws {
```
Comment (jaclync) [inline WooCommerce/WooCommerceTests/ViewRelated/Survey/SurveyViewControllerTests.swift:45]: nit: "does_not_completes" --> "does_not_complete"

**Comment by Garance91540** [comment]
Hi @Ecarrion, I know you probably are not done yet, but when you are close, I'd like to review the whole flow (prompt in the app, crowdsignal survey, feedback received screen) if you can open a PR for it?

**Comment by Ecarrion** [comment]
> Hi @Ecarrion, I know you probably are not done yet, but when you are close, I'd like to review the whole flow (prompt in the app, crowdsignal survey, feedback received screen) if you can open a PR for it?

Sure @Garance91540, I'll let you know!

**Comment by Ecarrion** [comment]
> I noticed there's a darker overlay right after opening the web view for a split second (like in your GIF), not sure what that flash was?

It was the default web view background color, I have updated it to "no-color"

**Pull request merged**